pkg/services/secrets/manager/manager.go

```diff
@@ -291,8 +291,15 @@ func newRandomDataKey() ([]byte, error) {
 }
 
 func (s *SecretsService) Decrypt(ctx context.Context, payload []byte) ([]byte, error) {
+	if len(payload) == 0 {
+		return nil, fmt.Errorf("unable to decrypt empty payload")
+	}
+
 	// Use legacy encryption service if featuremgmt.FlagDisableEnvelopeEncryption toggle is on
 	if s.features.IsEnabled(featuremgmt.FlagDisableEnvelopeEncryption) {
+		if len(payload) > 0 && payload[0] == '#' {
```

Comment:
I really like the idea of detecting & preventing an EE encrypted secret to be decrypted with legacy encryption 🔝 

I only have question about that:
- Can we guarantee secrets NOT encrypted with EE will never starts with '#'? 

I'm thinking that, alternatively, maybe we could try to add a bit more restrictive check, although that would be a bit more risky.

Additionally:
- Do you think it'd make sense to backport these changes to `v8.5`?


I think we've been able to reproduce the same issue (EE encrypted secrets decrypted with legacy encryption without apparent error) in v8.5 so, maybe it worths adding that check in the latest v8.5 🤷🏻  Any thoughts?

Thanks!

Reply:
>Can we guarantee secrets NOT encrypted with EE will never starts with '#'?

Given that we identify secret as the one encrypted by EE based on its first byte being '#', I think it's safe to assume none of legacy secrets should start with this sign.

Reply:
Also, if I get everything right, legacy have salt prepended to the ciphertext. Looking [at this code](https://github.com/grafana/grafana/blob/main/pkg/util/encoding.go#L19) the salt can't contain '#' as it's generated from alphanumeric characters.

@sakjur please, correct me if I'm wrong.

Reply:
Amazing, then 🚀